web/src/app/admin/AdminLimits.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useState } from 'react'
+import Button from '@material-ui/core/Button'
+import Card from '@material-ui/core/Card'
+import Grid from '@material-ui/core/Grid'
+import Typography from '@material-ui/core/Typography'
+import { makeStyles } from '@material-ui/core/styles'
+import gql from 'graphql-tag'
+import { startCase, isEmpty } from 'lodash-es'
+import AdminDialog from './AdminDialog'
+import PageActions from '../util/PageActions'
+import { Form } from '../forms'
+import AdminSection from './AdminSection'
+import { useQuery } from '@apollo/react-hooks'
+import Spinner from '../loading/components/Spinner'
+
+const query = gql`
+  query getLimits {
+    systemLimits {
+      id
+      description
+      value
+    }
+  }
+`
+const mutation = gql`
+  mutation($input: [SystemLimitInput!]!) {
+    setSystemLimits(input: $input)
+  }
+`
+
+const useStyles = makeStyles(theme => ({
+  gridContainer: {
+    [theme.breakpoints.up('md')]: {
+      justifyContent: 'center',
+    },
+  },
+  gridItem: {
+    [theme.breakpoints.up('md')]: {
+      maxWidth: '65%',
+    },
+  },
+  groupTitle: {
+    fontSize: '1.1rem',
+  },
+  saveDisabled: {
+    color: 'rgba(255, 255, 255, 0.5)',
+  },
+}))
+
+interface LimitsValues {
+  [id: string]: string
+}
+
+export default function AdminLimits() {
+  const classes = useStyles()
+  const [confirm, setConfirm] = useState(false)
+  const [values, setValues] = useState({})
+
+  const { data, loading } = useQuery(query)
+
+  if (loading && !data) {
+    return <Spinner />
+  }
```

Comment:
we should display an error if it fails to query: `const { data, error, loading } = useQuery(query)` and `if (error) return <GenericError error={error.message} />`